pkg/codegen/nodejs/gen.go

```diff
@@ -1098,15 +1082,15 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) (resourceFil
 				if comment == "" {
 					comment = fmt.Sprintf("The set of arguments for the %s.%s method.", name, method.Name)
 				}
-				if err := mod.genPlainType(w, methodName+"Args", comment, args, true, false, 1); err != nil {
+				if err := mod.genPlainType(exportedTypesWriter, methodName+"Args", comment, args, true, false, 1); err != nil {
 					return err
 				}
-				fmt.Fprintf(w, "\n")
+				fmt.Fprintf(exportedTypesWriter, "\n")
 			}
 		}
 
 		if fun.ReturnType != nil {
-			genReturnType := func(properties []*schema.Property) error {
+			genReturnType := func(properties []*schema.Property, w io.Writer) error {
```

Comment:
I don't think you need to do this, no? `w` is already in scope? Equally the rename is obscuring a bit whether this is critical to the PR working or not.